bin/build-nodes-json

```diff
@@ -63,11 +63,12 @@ def main():
         auth_url=auth_url
     )
 
+    bmc_id = [ s.id for s in nova.servers.list() if bmc_base in s.name ]
+    bm_id = [ s.id for s in nova.servers.list() if baremetal_base in s.name ]
+
     all_ports = sorted(neutron.list_ports()['ports'], key=lambda x: x['name'])
-    bmc_ports = list([p for p in all_ports
-                     if p['name'].startswith(bmc_base)])
-    bm_ports = list([p for p in all_ports
-                    if p['name'].startswith(baremetal_base)])
+    bmc_ports = [p for p in all_ports if p['device_id'] in bmc_id]
+    bm_ports = [p for p in all_ports if p['device_id'] in bm_id]
```

Comment:
We're going to need to filter bm_ports further or this won't support multiple baremetal networks.  I think the thing to do is to also filter bm_ports for only ports on private_net.

And actually there's another concern with this approach.  If there's no name set on the port, then the sort on line 69 isn't going to work sanely and we may end up with mismatched bmc and bm ports.  This may not be a big problem in CI, but when you're running this locally and want to see why a node isn't deploying it's really confusing if the bmcs don't match up right.

So you see why I'd rather just fix the bug that is causing the port not to attach correctly. :-)